src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -1869,6 +1869,19 @@ and solve_rigid_flex_or_flex_rigid_subtyping
               let combine t1 t2 wl =
                   let t1_base, p1_opt = base_and_refinement_maybe_delta false env t1 in
                   let t2_base, p2_opt = base_and_refinement_maybe_delta false env t2 in
+                  (*
+                   * AR: before applying op, we need to squash phi if required
+                   *     refinement formulas in F* may be in higher universe,
+                   *       meaning that if we apply op (l_and or l_or) directly, we may be
+                   *       unifying the universe of phi to zero, leading to errors
+                   *)
+                  let apply_op op phi1 phi2 =
+                    let squash phi =
+                      match env.universeof_fastpath env phi with
+                      | Some U_zero -> phi
+                      | Some u -> U.mk_squash u phi
+                      | None -> U.mk_squash (Env.new_u_univ ()) phi in
+                    op (squash phi1) (squash phi2) in
```

Comment:
cc @nikswamy , using `squash` when combining the refinements.

Reply:
nice catch